SwanCustomEnvironments/swancustomenvironments/serverextension.py

```diff
@@ -57,50 +50,24 @@ def _launch_makenv(self) -> subprocess.Popen:
         if nxcals:
             arguments.append("--nxcals")
 
-        return subprocess.Popen([self.makenv_path, *arguments], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-
-    async def _process_first_get(self, makenv_process) -> None:
-        """
-        Reads from the stdout of the makenv.sh process and sends it to the frontend.
-        This is done by the initial get request, which triggers the makenv.sh process.
-        """
-        SwanCustomEnvironmentsApiHandler.writing_log = True
-
-        # The first and any subsequent get requests will read from the process
-        # and stream out its output
         with open(self.LOG_FILE, "w") as log_file:
-            for line in iter(makenv_process.stdout.readline, b""):
-                line = line.decode("utf-8")
-                log_file.write(line)
+            # Create the environment asynchronously and send its output to the log file
+            return await asyncio.create_subprocess_exec(self.makenv_path, *arguments, stdout=log_file, stderr=log_file)
```

Comment:
Why does this need to become an `asyncio.subprocess.Process`? Doesn't it work the same with a regular `subprocess.Popen`? The latter is also asynchronous in the sense that it returns the process handle immediately and continues execution.